vector/src/main/java/im/vector/app/core/pushers/VectorMessagingReceiver.kt

```diff
@@ -0,0 +1,182 @@
+            notificationDrawerManager = notificationDrawerManager()
+            notifiableEventResolver = notifiableEventResolver()
+            pusherManager = pusherManager()
+            activeSessionHolder = activeSessionHolder()
+            vectorPreferences = vectorPreferences()
+        }
+    }
+
+    override fun onMessage(context: Context?, message: String, instance: String) {
+        initVar(context!!)
+        Timber.i("onMessage received")
+        lateinit var data: JSONObject
+        lateinit var notification: JSONObject
+        lateinit var eventId: String
+        try {
+            data = JSONObject(message)
+            notification = data.getJSONObject("notification")
+        } catch (e: JSONException) {
+            e.printStackTrace()
+            return
+        }
+        try {
+            eventId = notification.getString("event_id")
+        } catch (e: JSONException) {
+            Timber.i("No event_id on notification")
+            notification.put("event_id", "")
+            eventId = ""
+        }
+        try {
+            notification.getString("room_id")
+        } catch (e: JSONException) {
+            Timber.i("No room_id on notification")
+            notification.put("room_id", "")
+        }
+        if (eventId == PushersManager.TEST_EVENT_ID) {
+            val intent = Intent(NotificationUtils.PUSH_ACTION)
+            LocalBroadcastManager.getInstance(context).sendBroadcast(intent)
+            return
+        }
+
+        if (!vectorPreferences.areNotificationEnabledForDevice()) {
+            Timber.i("Notification are disabled for this device")
+            return
+        }
+
+        mUIHandler.post {
+            if (ProcessLifecycleOwner.get().lifecycle.currentState.isAtLeast(Lifecycle.State.STARTED)) {
+                // we are in foreground, let the sync do the things?
+                Timber.v("PUSH received in a foreground state, ignore")
+            } else {
+                onMessageReceivedInternal(context, notification)
+            }
+        }
+    }
+
+    override fun onNewEndpoint(context: Context?, endpoint: String, instance: String) {
+        initVar(context!!)
+        Timber.i("onNewEndpoint: adding $endpoint")
+        UPHelper.storeUpEndpoint(context, endpoint)
+        if (vectorPreferences.areNotificationEnabledForDevice() && activeSessionHolder.hasActiveSession()) {
+            val gateway = UPHelper.customOrDefaultGateway(context, endpoint)
+            UPHelper.storePushGateway(context, gateway)
+            UPHelper.storeUpEndpoint(context, endpoint)
+            pusherManager.registerPusher(endpoint, gateway)
+        }
+    }
+
+    override fun onRegistrationFailed(context: Context?, instance: String) {
+        Toast.makeText(context, "Registration Failed", Toast.LENGTH_SHORT).show()
+    }
+
+    override fun onRegistrationRefused(context: Context?, instance: String) {
+        Toast.makeText(context, "Registration Refused", Toast.LENGTH_SHORT).show()
```

Comment:
```suggestion
        Toast.makeText(context, "Push service registration refused by server", Toast.LENGTH_LONG).show()
```

(also check if `LENGTH_LONG` actually exists)